### ISSUE TITLE
creating a baseline after prototyping with db push doesn't work

### ISSUE THREAD

**Issue by danshapir** [prisma/prisma#8249]
### Bug description

After working for a day with db push, I wanted to create my first migration, this is impossible (happened in multiple different projects and is consistent).

I keep getting a Drift detected when trying to do px prisma migrate dev --name initial-migration --create-only but npx prisma db push returns all is synced. even did and introspect. I don’t have any migrations, I worked only with db push until now.

It just requires me to do a reset, which.. well, no?.. Haha

### How to reproduce

1. work with db push, prototype
2. Try and create a new initial migration
3. It will fail....

### Expected behavior

The baseline should be created

### Prisma information

With any schema it happened... Even after doing an introspect.

### Environment & setup

- OS:Mac
- Database:Postgres
- Node.js version: happened with 13-16

### Prisma Version

```
2.26
```

happened with previous version in the past month as well

**Comment by pantharshit00** [comment]
Hello

Can you please share an example schema which you are unable to baseline along with the error migrate which migrate throws?

Right now provided information is not enough for reproduction.

**Comment by danshapir** [comment]
It literally doesn't matter. Even the default schema with User, just add any other model and try. It will fail.

It basically never works.

**Comment by pantharshit00** [comment]
Indeed I can reproduce that. Classified this as confirmed bug.

**Comment by Jolg42** [comment]
Note from Harshit
>Looks like we are always detecting a drift after a db push execution
Before this screenshot I added Test table to the schema and ran db push.
![image (31)](https://user-images.githubusercontent.com/1328733/128016075-99f5715c-4a07-48d4-9b79-b2a1606ff9b5.png)

[Internal Slack thread](https://prisma-company.slack.com/archives/CGMCB82N8/p1627633792002600)

**Comment by thebiglabasky** [comment]
The problem there is related to how `migrate` and `db push` are designed to work.

In this very case, the experience is clearly suboptimal and we should figure out how to improve it (better error messaging, new feature...).

Requalifying as a feature now.

**Comment by danshapir** [comment]
@thebiglabasky Well the main documentation states clearly, start with db push, get to some good structure and then do a baseline.
It makes no sense to start with migrations from the get go, as this means there will be dozens of add column, drop column, change type, add index different migrations in the beginning.

So I find it extremely critical, as this happens just on your baby steps.

**Comment by thebiglabasky** [comment]
Indeed. Hence my point about it being clearly suboptimal: we'll look into prioritizing that along with the other initiatives we're looking at.
